File: home_assignments/java/TicketSeller/TicketSellerModel/src/main/java/ticketseller/model/validators/implementations/UserValidator.java
package ticketseller.model.validators.implementations;


import ticketseller.model.entities.User;
import ticketseller.model.validators.ValidationException;
import ticketseller.model.validators.interfaces.IUserValidator;

public class UserValidator implements IUserValidator {
    @Override
    public void validate(User user) throws ValidationException {
    }

    @Override
    public void validateForLogin(String username, String password) throws ValidationException {
        String error = "";
        if (username == null)
            error += "Username is null;\n";
        if (password == null)
            error += "Password is null;\n";
        if (!error.equals(""))
            throw new ValidationException(error);
    }
}

File: home_assignments/csharp/TicketSeller/model/validators/implementations/UserValidator.cs
using System;
using model.entities;
using model.validators.interfaces;

namespace model.validators.implementations;

public class UserValidator:IUserValidator
{
    public void Validate(User entity)
    {
        throw new NotImplementedException();
    }
    public void ValidateForLogin(string username, string password)
    {
        String error = "";
        if (username == null)
            error += "Username is null;\n";
        if (password == null)
            error += "Password is null;\n";
        if (!error.Equals(""))
            throw new ValidationException(error);
    }
}